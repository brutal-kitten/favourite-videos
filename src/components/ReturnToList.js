import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ReturnToList extends Component {

  static propTypes = {
    returnToList: PropTypes.func.isRequired
  }

  //calls function that shows user the full list instead of favorites
  handleClick = (event) => {

    event.preventDefault();
    this.props.returnToList();
  }

  render () {

    return(
      <div className="returnToList" >
        <button tabIndex='0' className="buttonReturnToList" type="button" onClick={(event) => this.handleClick(event)}>Return to list</button>
      </div>
    )
  }
}

export default ReturnToList;
