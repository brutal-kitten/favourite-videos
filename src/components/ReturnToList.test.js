import React from 'react';
import ReactDOM from 'react-dom';
import ReturnToList from './ReturnToList.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('should display a button', () => {
  const tree = TestRenderer.create(
    <ReturnToList />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe ('shoud call handleClick function after click', () => {
  const mockReturnToList = jest.fn();
  const props = {
    returnToList: mockReturnToList,
};

  const wrapper = shallow(<ReturnToList {...props}/>);

  wrapper.find('button').simulate('click', {
    preventDefault: () => {}
  });

  it('should call returnToList function', () => {
    expect(mockReturnToList.mock.calls.length).toBe(1);
  });

})
