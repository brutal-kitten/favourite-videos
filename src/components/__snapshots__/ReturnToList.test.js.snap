// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a button 1`] = `
<div
  className="returnToList"
>
  <button
    className="buttonReturnToList"
    onClick={[Function]}
    tabIndex="0"
    type="button"
  >
    Return to list
  </button>
</div>
`;
